hack/build-platform.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Build all compile targets and the base binaries for only the platform of the
+# host it's building on. (i.e. - no cross compile)
+STARTTIME=$(date +%s)
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
+
+source <(go env)
+platforms=( "${GOHOSTOS}/${GOARCH}" )
+if [[ -n "${OS_ONLY_BUILD_PLATFORMS-}" ]]; then
+  filtered=()
+  for platform in ${platforms[@]}; do
+    if [[ "${platform}" =~ "${OS_ONLY_BUILD_PLATFORMS}" ]]; then
+      filtered+=("${platform}")
+    fi
+  done
+  platforms=("${filtered[@]}")
+  echo "DEBUGGING: ${filtered[@]}"
+fi
+
+# Build the primary client/server for the host platform
+OS_BUILD_PLATFORMS=("${platforms[@]}")
+if [[ "${GOHOSTOS}/${GOARCH}" == "linux/amd64" ]]; then
+    OS_GOFLAGS_LINUX_AMD64="-tags=gssapi" os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}"
+elif [[ "${GOHOSTOS}/${GOARCH}" == "linux/arm64" ]]; then
+    OS_GOFLAGS_LINUX_ARM64="-tags=gssapi" os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}"
+elif [[ "${GOHOSTOS}/${GOARCH}" == "linux/386" ]]; then
+    OS_GOFLAGS_LINUX_386="-tags=gssapi" os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}"
+elif [[ "${GOHOSTOS}/${GOARCH}" == "linux/arm" ]]; then
+    OS_GOFLAGS_LINUX_ARM="-tags=gssapi" os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}"
+elif [[ "${GOHOSTOS}/${GOARCH}" == "linux/ppc64" ]]; then
+    OS_GOFLAGS_LINUX_PPC64="-tags=gssapi" os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}"
+elif [[ "${GOHOSTOS}/${GOARCH}" == "linux/ppc64le" ]]; then
+    OS_GOFLAGS_LINUX_PPC64LE="-tags=gssapi" os::build::build_binaries "${OS_CROSS_COMPILE_TARGETS[@]}"
+fi
+
+# Pass the necessary tags
+OS_GOFLAGS="${OS_GOFLAGS:-} ${OS_IMAGE_COMPILE_GOFLAGS}" os::build::build_static_binaries "${OS_IMAGE_COMPILE_TARGETS[@]-}" "${OS_SCRATCH_IMAGE_COMPILE_TARGETS[@]-}"
```

Comment:
It's a nuanced point, I believe we almost always want the `:-` behavior versus the `-` behavior. We should fix it everywhere, but I wouldn't put that in the scope of this PR.